src/hotspot/share/gc/shared/objectCountClosure.hpp

```diff
@@ -0,0 +1,24 @@
+#ifndef SHARE_GC_SHARED_OBJECTCOUNTCLOSURE_HPP
+#define SHARE_GC_SHARED_OBJECTCOUNTCLOSURE_HPP
+
+#include "gc/shared/gcId.hpp"
+#include "gc/shared/objectCountEventSender.hpp"
+#include "jfr/jfrEvents.hpp"
+#include "memory/heapInspection.hpp"
+#include "utilities/macros.hpp"
+#include "utilities/ticks.hpp"
+
+class KlassInfoEntry;
+class Klass;
+
+class ObjectCountClosure : public ObjectCountEventSender {
+    static KlassInfoTable cit;
+
+public:
+    static bool check_table_exists();
```

Comment:
Instead of constructing and destructing the table(s) each time, consider if it might make sense to keep pre-allocated table and merely clearing entries as you emit event for each entry.